### PR TITLE
Add ContextMenu to beatmap external link

### LINKED ISSUE #8715: Add a button “Copy URL-map” for quick sending to chat
**Describe the new feature:** 
I think it can be added near the "open in browser" button, because it will allow sharing maps in the lazer itself (the command /np does not work yet, although this function can work with the /np command)
Maybe add the command /c for this action?
osu_2020-04-11_21-10-21

### PR DESCRIPTION
**Pull request by andrewhong04**
Fixes: https://github.com/ppy/osu/issues/8715

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/ExternalLinkButton.cs [hunk 1]
```diff
@@ -42,6 +49,31 @@ public ExternalLinkButton(string link = null)
             };
         }
 
+        private void copyUrl()
+        {
+            host.GetClipboard()?.SetText(Link);
+            notificationOverlay.Post(new SimpleNotification
+            {
+                Text = "Copied URL!"
+            });
```
Comment (frenzibyte) [inline osu.Game/Graphics/UserInterface/ExternalLinkButton.cs:58]: A notification shouldn't be shown when copying a URL, that would only be annoying to the user.

File: osu.Game/Graphics/UserInterface/ExternalLinkButton.cs [hunk 2]
```diff
@@ -42,6 +49,31 @@ public ExternalLinkButton(string link = null)
...
+        public MenuItem[] ContextMenuItems
+        {
+            get
+            {
+                List<MenuItem> items = new List<MenuItem>();
+
+                if (Link != null)
+                {
+                    items.Add(new OsuMenuItem("Open", MenuItemType.Standard, () => host.OpenUrlExternally(Link)));
+                    items.Add(new OsuMenuItem("Copy URL", MenuItemType.Standard, () => copyUrl()));
```
Comment (frenzibyte) [inline osu.Game/Graphics/UserInterface/ExternalLinkButton.cs:70]: As per the inspection:
```suggestion
                    items.Add(new OsuMenuItem("Copy URL", MenuItemType.Standard, copyUrl));
```

File: osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs [hunk 3]
```diff
@@ -90,113 +91,117 @@ public BeatmapSetHeaderContent()
...
-                        AutoSizeAxes = Axes.Y,
-                        Padding = new MarginPadding
-                        {
-                            Vertical = BeatmapSetOverlay.Y_PADDING,
-                            Left = BeatmapSetOverlay.X_PADDING,
-                            Right = BeatmapSetOverlay.X_PADDING + BeatmapSetOverlay.RIGHT_WIDTH,
-                        },
-                        Children = new Drawable[]
+                        RelativeSizeAxes = Axes.Both,
+                        Child = new Container
```
Comment (frenzibyte) [inline osu.Game/Overlays/BeatmapSet/BeatmapSetHeaderContent.cs:97]: This should not use `RelativeSizeAxes = Axes.Both` in this hierarchy, look at the sizing specifications of the parent of this container and the child here.

For reference, this is how this looks now on a beatmap with many difficulties:

| master | PR |
|--------|-------|
| CleanShot 2022-07-26 at 10 42 49@2x | CleanShot 2022-07-26 at 10 41 17@2x |

**Review by frenzibyte — CHANGES_REQUESTED** [review]

**Review by frenzibyte — APPROVED** [review]
Seems fine, will wait for a second review UX-wise.

**Review by smoogipoo — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**